ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/DataConnections.resource

```diff
@@ -160,3 +160,26 @@ Get Data Connection Form Values
     ${s3_region}   get element attribute    ${S3_REGION_DC_INPUT_XP}      value
     ${s3_bucket}   get element attribute    ${S3_BUCKET_DC_INPUT_XP}      value
     [return]    ${s3_name}    ${s3_key}    ${s3_secret}    ${s3_endpoint}    ${s3_region}    ${s3_bucket}
+
+Recreate S3 Data Connection
+    [Documentation]    Edit Exisisting Or Create New S3 Data Connection from DS Project details page
+    [Arguments]    ${project_title}    ${dc_name}    ${aws_access_key}    ${aws_secret_access}    ${new_dc_name}=${dc_name}
+    ...            ${aws_s3_endpoint}=${S3.AWS_DEFAULT_ENDPOINT}    ${aws_region}=${S3.AWS_DEFAULT_REGION}
+    ...            ${connected_workbench}=${NONE}    ${press_cancel}=${FALSE}    ${aws_bucket_name}=${NONE}
+    ...            ${aws_s3_ca_bundle}=${NONE}
+    ${is_exist} =  Run Keyword And Return Status
+        ...    Wait Until Page Contains Element    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/*/div[text()="${dc_name}"]]
+    IF  "${is_exist}" == "${TRUE}"   
+        Log    Data Connection '${dc_name}' already exists, reusing connection    console=True
```

Comment:
it is not re-using but it's editing, I think it's different and should be clear when using the kw